# PISA 2012 Dataset Exploration 
## by Maryse Bonhomme

## Preliminary Wrangling

> PISA is the OECD's Program for International Students Assessment. The dataset that will be used today for analysis is the PISA 2012 survey, assessing 15 years old from 65 countries in  reading, mathematics and science.  And ultimately measure how well, those children are prepared for the future. <br>
The PISA survey is conducted every 3 years. we are using the responses from the PISA 2012 survey. <br>
That survey covered many areas of the 15 years old life, asking questions about life at school and also outside school.<br>
The results this huge dataset with more than 600  variables and responses from almost 510, 000 students ( 485490 rows are part of this dataset).

In [21]:
# import all packages and set plots to be embedded inline
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
sb.set_style("darkgrid")

%matplotlib inline

> *Load in your dataset and describe its properties through the questions below.
Try and motivate your exploration goals through this section.
The dataset covers a broad areas of the 15 years old. As part of the analysis today, I will focus  on the following questions:*
* How does do Students who uses the internet at home differs from children who don't use internet at home in terms of scores? 
* How do children who reported playing video games compares to the one who don't? 
* I was also very interested by the cognitive activation section of the survey. how do children who reports being always encouraged to finding their own formulation to problems compares to those who are not really encouraged to do so?
* finally, how do non OECD and OECD countries perform in Math, science and reading? 
* Do boys and girls perform in the same ways on test, in OECD and non-OECD countries? 
<hr>
* what is the distribution of scores in Math, reading and science <br>
* Which countries perform best in each category?  <br>
* since in real life, the ability to adapt to new context is crucial, I was very interested in the cognitive activation section of the survey. in which countries are children most encouraged to find to think on their own, are given the freedom to explore different ways of finding answers. <br>

of course this is a survey and what we are measuring is the perception those 15 years old have of being encouraged to  think more deeply about what they have learning in school. and ultimalty might make them more intelligent. 
In many educational system, academy success does not depend only on the ability to understand but also on other abilities like the ability to memorize for example. <br>
I am personally convinced that in real life, that ability is crucial. but in this analysis, I am interested in understanding if there is any correlation with scores. 


In [22]:
#loading the dataset
pd.set_option('display.max_columns', None)
df = pd.read_csv('pisa2012.csv',encoding='cp1252')

C:\Users\gwiza\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2785: DtypeWarning: Columns (15,16,17,21,22,23,24,25,26,30,31,36,37,45,65,123,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,284,285,286,287,288,289,290,291,292,293,294,295,296,297,298,299,300,301,302,303,307,308,309,310,311,312,313,314,315,316,317,318,319,320,321,322,323,324,325,326,327,328,329,330,331,332,333,334,335,336,337,338,339,340,341,342,343,344,345,346,347,348,349,350,351,352,353,354,355,356,357,376,377,378,379,380,381,382,383,384,385,386,387,388,389,390,391,392,393,394,395,396,397,398,399,400,401,402,403,475) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [23]:
# using panda's sample to visually check the data
df.sample(10)

Unnamed: 0                   CNT  SUBNATIO  STRATUM      OECD  \
473835      473836               Uruguay   8580000  URY0005  Non-OECD   
463487      463488               Tunisia   7880000  TUN0002  Non-OECD   
30899        30900             Australia    360000  AUS0617      OECD   
408403      408404         Florida (USA)   8400100  QUA9797  Non-OECD   
57415        57416                Brazil    760000  BRA1130  Non-OECD   
10093        10094  United Arab Emirates   7840100  ARE0101  Non-OECD   
141961      141962               Denmark   2080000  DNK0003      OECD   
42652        42653               Belgium    560100  BEL0108      OECD   
434426      434427       Slovak Republic   7030000  SVK0015      OECD   
55627        55628                Brazil    760000  BRA1750  Non-OECD   

                               NC  SCHOOLID  STIDSTD  ST01Q01  ST02Q01  \
473835                    Uruguay       116     3598       10      6.0   
463487                    Tunisia        89     2505       10      2.0   
30899                   Australia       472     8749       11      5.0   
408403  United States of America         36     1249       10      1.0   
57415                     Brazil        104     2150       10      1.0   
10093       United Arab Emirates        210     5351        8      1.0   
141961                    Denmark       178     3860        9      1.0   
42652                     Belgium        47     1266       10      5.0   
434426            Slovak Republic        71     1421       10      6.0   
55627                     Brazil         17      362       11      1.0   

        ST03Q01  ST03Q02 ST04Q01                      ST05Q01  ST06Q01  \
473835        7     1996  Female  Yes, for more than one year      6.0   
463487        3     1996  Female   Yes, for one year or less       6.0   
30899         6     1996    Male  Yes, for more than one year      6.0   
408403        2     1997  Female   Yes, for one year or less       6.0   
57415         1     1996    Male                         No        8.0   
10093         2     1996  Female   Yes, for one year or less       6.0   
141961       12     1996  Female  Yes, for more than one year      5.0   
42652        10     1996  Female  Yes, for more than one year      6.0   
434426        7     1996    Male  Yes, for more than one year      6.0   
55627         4     1996  Female  Yes, for more than one year      7.0   

          ST07Q01    ST07Q02    ST07Q03                ST08Q01  \
473835  No, never  No, never  No, never     One or two times     
463487  No, never  No, never  No, never                 None     
30899   No, never  No, never  No, never                 None     
408403  No, never  No, never  No, never  Three or four times     
57415   No, never  No, never  No, never                 None     
10093   Yes, once  Yes, once        NaN   Five or more times     
141961  No, never  No, never        NaN     One or two times     
42652   No, never  No, never  No, never     One or two times     
434426  No, never  No, never  No, never                 None     
55627         NaN        NaN        NaN                 None     

                      ST09Q01  ST115Q01 ST11Q01 ST11Q02 ST11Q03 ST11Q04  \
473835                 None         1.0     Yes      No      No      No   
463487                 None         1.0     Yes     Yes     Yes     Yes   
30899                  None         1.0     Yes     Yes     Yes     Yes   
408403     One or two times         1.0     Yes      No     Yes      No   
57415      One or two times         2.0     NaN     NaN     Yes     NaN   
10093    Five or more times         3.0     Yes     Yes     Yes     Yes   
141961                 None         1.0     Yes     Yes     Yes     Yes   
42652                  None         1.0     Yes     Yes     NaN     Yes   
434426                 None         1.0     Yes     NaN     NaN     NaN   
55627   Three or four times         1.0     NaN     NaN     NaN     NaN   

       ST11Q05 ST11Q0

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 485490 entries, 0 to 485489
Columns: 636 entries, Unnamed: 0 to VER_STU
dtypes: float64(250), int64(18), object(368)
memory usage: 2.3+ GB


In [25]:
df.shape

(485490, 636)

### What is the structure of your dataset?

> The PISA dataset is a dataset with 636 columns and 495 489 rows. 

### What is/are the main feature(s) of interest in your dataset?

> **_Video Game and Internet users Score_**: With the internet and video games being constantly critized for kids, I am really interested in observing if there any correlation between using the internet or playing video games and scores obtained. <br>
> **_In term of cognitive Activation_**: Is there any correlation between children who are encouraged to think for themselves about problems, instead of being asking questions with obvious questions. <br>
> **_Scores _**:Is there a difference in averages scores, for OECD and non OECD countries? 

### What features in the dataset do you think will help support your investigation into your feature(s) of interest?

> In order to find answers to those questions, I will keep informations on countries, respondents gender, cognitive activations questions and of course score. without forgetting responses on internet and video games. 
Below I am going to reduce the  original dataframe to only the column needed. 

In [26]:
#selecting wanted_columns = ['CNT', 'OECD', 'ST01Q01', 'ST02Q01', 'ST04Q01','ST28Q01','ST53Q02','ST80Q01','ST80Q04',
                  'ST80Q05','ST80Q07', 'ST80Q09', 'ST87Q01','ST87Q02','ST87Q04','ST87Q05', 'ST87Q07','IC01Q02','IC01Q03',
                  'IC01Q04','IC01Q05', 'IC01Q06', 'IC01Q07', 'PV1MATH', 'PV2MATH', 'PV3MATH', 'PV4MATH', 'PV5MATH', 'PV1READ',
                  'PV2READ', 'PV3READ', 'PV4READ', 'PV5READ', 'PV1SCIE', 'PV2SCIE', 'PV3SCIE','PV4SCIE', 'PV5SCIE']

IndentationError: unexpected indent (<ipython-input-26-5679818d9a10>, line 2)

In [ ]:
df_selected = df[wanted_columns].copy()

In [ ]:
df_selected.sample(5)

In [ ]:
df_selected.rename(columns=
                   {'CNT':'country', 
                    'OECD':'OECD', 
                    'ST01Q01':'international_grade',
                    'ST02Q01':'national_study_program', 
                    'ST04Q01':'gender',
                    'ST28Q01':'number_books',
                    'ST53Q02':'learning_strategies',
                    'ST80Q01':'CA_encouraged_reflect',
                    'ST80Q04':'CA_required_to_think',
                    'ST80Q05':'CA_use_own_procedures',
                    'ST80Q07':'CA_presents_problems_different_context',
                    'ST80Q09':'CA_asks_for_explanations',
                    'ST87Q01':'SB_feel_like_outsider',
                    'ST87Q02':'SB_make_friends_easily',
                    'ST87Q04':'SB_feel_awkward_at_school',
                    'ST87Q05':'SB_liked_by_other_students',
                    'ST87Q07':'SB_feel_happy_at_School',
                    'IC01Q02':'home_laptop',
                    'IC01Q03':'home_tablet',
                    'IC01Q04':'home_internet',
                    'IC01Q05':'home_video_games_console',
                    'IC01Q06':'home_cell_phone_no_internet',
                    'IC01Q07':'home_cell_phone_with_internet'}, inplace=True)

In [ ]:
df_selected.head(1)

## Univariate Exploration

> In this section, investigate distributions of individual variables. If
you see unusual points or outliers, take a deeper look to clean things up
and prepare yourself to look at relationships between variables.

### Math Score Exploration
I am going to explore Math score, but Before I proceed, I am going to first make the average of the 5 plausible values available. 

In [ ]:
df_selected['math_average'] =  (df_selected['PV1MATH'] + df_selected['PV2MATH']+ 
                                df_selected['PV3MATH'] + df_selected['PV4MATH'] + df_selected['PV5MATH'])/5
df_selected['reading_average'] = (df_selected ['PV1READ'] + df_selected['PV2READ']+ 
                          df_selected['PV3READ'] + df_selected ['PV4READ'] + df_selected['PV5READ'])/5
df_selected['science_average'] = (df_selected['PV1SCIE'] + df_selected ['PV2SCIE']+ 
                          df_selected['PV3SCIE'] + df_selected['PV4SCIE'] + df_selected['PV5SCIE'])/5


In [28]:
df_selected.head(1)

country      OECD  international_grade  national_study_program  gender  \
0  Albania  Non-OECD                   10                     1.0  Female   

  number_books learning_strategies    CA_encouraged_reflect  \
0  0-10 books                  NaN  Always or almost always   

  CA_required_to_think CA_use_own_procedures  \
0            Sometimes       Never or rarely   

  CA_presents_problems_different_context CA_asks_for_explanations  \
0                Always or almost always  Always or almost always   

  SB_feel_like_outsider SB_make_friends_easily SB_feel_awkward_at_school  \
0                 Agree                  Agree            Strongly agree   

  SB_liked_by_other_students SB_feel_happy_at_School home_laptop home_tablet  \
0             Strongly agree                   Agree         NaN         NaN   

  home_internet home_video_games_console home_cell_phone_no_internet  \
0           NaN                      NaN                         NaN   

  home_cell_phone_with_internet   PV1MATH   PV2MATH   PV3MATH   PV4MATH  \
0                           NaN  406.8469  376.4683  344.5319  321.1637   

    PV5MATH   PV1READ  PV2READ   PV3READ   PV4READ   PV5READ   PV1SCIE  \
0  381.9209  249.5762  254.342  406.8496  175.7053  218.5981  341.7009   

   PV2SCIE   PV3SCIE   PV4SCIE   PV5SCIE  math_average  reading_average  \
0   408.84  348.2283  367.8105  392.9877     366.18634        261.01424   

   science_average  
0        371.91348

In [ ]:
df_selected['math_average'].describe()

In [ ]:
sb.countplot(data = df_selected, x = 'math_average')

> Make sure that, after every plot or related series of plots, that you
include a Markdown cell with comments about what you observed, and what
you plan on investigating next.

### Discuss the distribution(s) of your variable(s) of interest. Were there any unusual points? Did you need to perform any transformations?

> Your answer here!

### Of the features you investigated, were there any unusual distributions? Did you perform any operations on the data to tidy, adjust, or change the form of the data? If so, why did you do this?

> Your answer here!

## Bivariate Exploration

> In this section, investigate relationships between pairs of variables in your
data. Make sure the variables that you cover here have been introduced in some
fashion in the previous section (univariate exploration).

### Talk about some of the relationships you observed in this part of the investigation. How did the feature(s) of interest vary with other features in the dataset?

> Your answer here!

### Did you observe any interesting relationships between the other features (not the main feature(s) of interest)?

> Your answer here!

## Multivariate Exploration

> Create plots of three or more variables to investigate your data even
further. Make sure that your investigations are justified, and follow from
your work in the previous sections.

### Talk about some of the relationships you observed in this part of the investigation. Were there features that strengthened each other in terms of looking at your feature(s) of interest?

> Your answer here!

### Were there any interesting or surprising interactions between features?

> Your answer here!

> At the end of your report, make sure that you export the notebook as an
html file from the `File > Download as... > HTML` menu. Make sure you keep
track of where the exported file goes, so you can put it in the same folder
as this notebook for project submission. Also, make sure you remove all of
the quote-formatted guide notes like this one before you finish your report!

## References:
* PISA 2012 contest winner: http://mi2.mini.pw.edu.pl:8080/SmarterPoland/PISAcontest/
* What is PISA: http://www.oecd.org/pisa/
* OECD: https://en.wikipedia.org/wiki/OECD
* PISA 2012 Countries : http://www.oecd.org/pisa/aboutpisa/pisa-2012-participants.htm
* PISA 2012 Data analysis Manual:http://archivos.agenciaeducacion.cl/Manual_de_Analisis_de_datos_SPSS_version_ingles.pdf
* PISA Data Exploration : https://github.com/brunoassisp/PISA-Data-Viz-project/blob/master/PISA_EDA.ipynb
* Jasccyang github repository : https://github.com/jasccyang/PISA2012/blob/master/Reducing%20File%20Size.ipynb
* Seby Sbirna Github repository: https://github.com/seby-sbirna/Data-Analyst-Nanodegree-Project-Portfolio/blob/master/Project%205%20-%20Communicate%20Data%20Findings/Project%205%20-%20Sebastian%20Sbirna.ipynb